source/features/highlight-file-in-prs.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getOwnerAndRepo, getRepoURL} from '../libs/utils';
+import {isSingleFile} from '../libs/page-detect';
+import getDefaultBranch from '../libs/get-default-branch';
+import {groupSiblings} from '../libs/group-buttons';
+// TODO: implement caching:
+// import cache from 'webext-storage-cache';
+import * as icons from '../libs/icons';
+
+async function init(): Promise<void> {
+	const {ownerName, repoName} = getOwnerAndRepo();
+	const defaultBranch = await getDefaultBranch();
+	const cacheKey = `highlight-file-in-prs:${ownerName}/${repoName}`;
+	console.log(cacheKey, defaultBranch, getRepoURL());
+
+	// TODO: replace first (2x) with 100.
+	const result = await api.v4(
+		`{
+			repository(owner: "${ownerName}", name: "${repoName}") {
+				pullRequests(first: 2, states: OPEN, baseRefName: "${defaultBranch}") {
+					nodes {
+						number
+						files(first: 3) {
+							nodes {
+								path
+							}
+						}
+					}
+				}
+			}
+		}`
+	);
+	console.log('result', result, result.repository.pullRequests.nodes);
+
+	const files: any = {};
+	for (const node of result.repository.pullRequests.nodes) {
+		console.log('node', node);
+		for (const file of node.files.nodes) {
+			console.log('file', file);
+			if (!files[file.path]) {
+				files[file.path] = [];
+			}
+
+			files[file.path].push(node.number);
+		}
+	}
+
+	console.log('files', files);
+
+	const path = select<HTMLElement>('clipboard-copy, #blob-edit-path')!.getAttribute('value')!; // `clipboard-copy` on blob page, `#blob-edit-path` on edit page.
+	console.log('path', path);
+	if (!files[path]) {
+		return;
+	}
+
+	const wrapper = <div className="rgh-highlight-file-in-prs"></div>;
```

Comment:
You can use self-closing tags if they’re empty
```suggestion
	const wrapper = <div className="rgh-highlight-file-in-prs" />;
```

Reply:
I was thinking of appending the buttons in here, but went with the for..of loop.